### PR TITLE
React Hook Form

### PR DESCRIPTION
**Pull request by a-reznikov**
image
image

- [x] Implement a Modal
- [x] Use React Hook Form
- [x] Create API Routes

### CODE REVIEW

File: src/components/common/subscription-form/SubscriptionForm.tsx [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+              <label htmlFor="female" className="block text-sm font-medium">
+                Female
+              </label>
+            </div>
+          </div>
+        </fieldset>
+      </div>
+      <button
+        className="text-white px-2 py-1 rounded bg-main max-w-fit self-end"
+        type="submit"
```
Comment (Ilya-Valasiuk) [inline src/components/common/subscription-form/SubscriptionForm.tsx:178]: do we have disables state?

File: src/api/subscription/queries.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { queryClient } from "@/constants/query-client";
+import { setSubscribe } from "./module";
+import { SubscriptionFormData } from "./types";
+
+enum QueryKeys {
+  SUBSCRIBE = "subscribe",
+}
+
+export const fetchSubscribe = async (data: SubscriptionFormData) =>
```
Comment (Ilya-Valasiuk) [inline src/api/subscription/queries.ts]: for POST methods you don't need useQuery. Please read about mutations https://tanstack.com/query/v4/docs/framework/react/guides/mutations

Comment (a-reznikov) [reply src/api/subscription/queries.ts]: Used mutation

File: src/components/common/subscription-form/SubscriptionForm.tsx (hunk 1 above)
Comment (a-reznikov) [reply src/components/common/subscription-form/SubscriptionForm.tsx:178]: Implemented reset form state and disable button while values are not valid

**Pull request merged**